woocommerce/woocommerce
# fix PHP 8 warning in Blocks\Installer\create_pages() callback when oher filters have removed pages

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

PHP 8 warnings for `Undefined array key "cart"` are thrown when other code removes the cart or checkout pages via the `woocommerce_create_pages` filter.  This simply validates that those keys exist prior to attempting to access them.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. While running PHP8+ Add a filter like the one below in another plugin or theme that is active:
```
add_filter( 'woocommerce_create_pages','__return_empty_array', 9 );
```
2. Install WooCommerce
3. Check logs for warnings.
